### PR TITLE
Update tendermint-rs to v0.38

### PR DESCRIPTION
**Pull request by yito88**
## Describe your changes
- Update tendermint-rs, ibc-rs (based on v0.54), prost, and tower-abci

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes

### CODE REVIEW

File: crates/ibc/src/context/common.rs [hunk 1]
```diff
@@ -222,7 +222,13 @@ pub trait IbcCommonContext: IbcStorageContext {
                     "Decoding the client update time failed: ID {client_id}",
                 ),
             })?
-            .into();
+            .try_into()
+            .map_err(|_| ClientError::Other {
+                description: format!(
+                    "Convertsion of the client update time failed: ID \
```
Comment (tzemanovic) [inline crates/ibc/src/context/common.rs:228]:
```suggestion
                    "Conversion of the client update time failed: ID \
```

**Review by tzemanovic — APPROVED** [review]

**Comment by tzemanovic** [comment]
@mergifyio requeue

**Comment by Fraccaman** [comment]
@mergify queue

**Comment by tzemanovic** [comment]
@mergifyio requeue

**Comment by brentstone** [comment]
@mergifyio requeue

**Pull request merged**